arch/arm/mach-omap/am335x/mcasp_am335x.c

```diff
@@ -0,0 +1,1423 @@
+
+	inst_arg = argv[0];
+
+	ret = sscanf(inst_arg, "mcasp%u", &instance);
+	if (ret != 1) {
+		help(HELP_MCASP);
+		return OMAPCONF_ERR_ARG;
+	}
+
+	ret = am335x_mcasp_is_clocked(instance);
+	if (ret < 0) {
+		fprintf(stderr, "omapconf: %s is not a valid instance\n",
+			inst_arg);
+		dprintf("omapconf: no clock information for %s\n", inst_arg);
+		help(HELP_MCASP);
+		return OMAPCONF_ERR_ARG;
+	} else if (ret == 0) {
+		fprintf(stderr, "omapconf: %s is not active, cannot proceed further\n",
+			inst_arg);
+		return 0;
+	}
+
+	baseaddr = am335x_mcasp_get_baseaddr(instance);
+	if (!baseaddr) {
+		fprintf(stderr, "omapconf: %s is not a valid instance\n",
+			inst_arg);
+		dprintf("omapconf: no base address for %s\n", inst_arg);
+		help(HELP_MCASP);
+		return OMAPCONF_ERR_NOT_AVAILABLE;
+	}
+
+	serializers = am335x_mcasp_get_serializers(instance);
+	if (serializers < 0) {
+		fprintf(stderr, "omapconf: %s is not a valid instance\n",
+			inst_arg);
+		dprintf("omapconf: no serializers for %s\n", inst_arg);
+		help(HELP_MCASP);
+		return OMAPCONF_ERR_NOT_AVAILABLE;
+	}
+
+	dprintf("%s(): McASP%u baseaddr=0x%08x num_ser=%u\n",
+		__func__, instance, baseaddr, serializers);
+
+	regs->num_serializers = serializers;
+
+	MCASP_READREG(baseaddr, regs->pwridlesysconfig);
+	MCASP_READREG(baseaddr, regs->pfunc);
+	MCASP_READREG(baseaddr, regs->pdir);
+	MCASP_READREG(baseaddr, regs->pdout);
+	MCASP_READREG(baseaddr, regs->pdin);
+	MCASP_READREG(baseaddr, regs->gblctl);
+	MCASP_READREG(baseaddr, regs->rxmask);
+	MCASP_READREG(baseaddr, regs->rxfmt);
+	MCASP_READREG(baseaddr, regs->rxfmctl);
+	MCASP_READREG(baseaddr, regs->aclkrctl);
+	MCASP_READREG(baseaddr, regs->ahclkrctl);
+	MCASP_READREG(baseaddr, regs->rxtdm);
+	MCASP_READREG(baseaddr, regs->evtctlr);
+	MCASP_READREG(baseaddr, regs->rxstat);
+	MCASP_READREG(baseaddr, regs->rxtdmslot);
+	MCASP_READREG(baseaddr, regs->revtctl);
+	MCASP_READREG(baseaddr, regs->txmask);
+	MCASP_READREG(baseaddr, regs->txfmt);
+	MCASP_READREG(baseaddr, regs->txfmctl);
+	MCASP_READREG(baseaddr, regs->aclkxctl);
+	MCASP_READREG(baseaddr, regs->ahclkxctl);
+	MCASP_READREG(baseaddr, regs->txtdm);
+	MCASP_READREG(baseaddr, regs->evtctlx);
+	MCASP_READREG(baseaddr, regs->txstat);
+	MCASP_READREG(baseaddr, regs->txtdmslot);
+	MCASP_READREG(baseaddr, regs->xevtctl);
+	MCASP_READREG(baseaddr, regs->wfifoctl);
+	MCASP_READREG(baseaddr, regs->wfifosts);
+	MCASP_READREG(baseaddr, regs->rfifoctl);
+	MCASP_READREG(baseaddr, regs->rfifosts);
+	for (i = 0; i < regs->num_serializers; i++)
+		MCASP_READREG(baseaddr, regs->xrsrctl[i]);
+
+	am335x_mcasp_data(regs, stream);
+	am335x_mcasp_control(regs, stream);
+	am335x_mcasp_clocking(regs, stream);
+	am335x_mcasp_frame_sync(regs, stream);
+	am335x_mcasp_format_units(regs, stream);
+	am335x_mcasp_serializers(regs, stream);
+	am335x_mcasp_pinctrl(regs, stream);
+
+	return 0;
+}
+
+/* ------------------------------------------------------------------------*//**
+ * @FUNCTION		am335xxx_mcasp_main
```

Comment:
s/am335xxx/am335x/g

Reply:
>there will be duplication of the register 'dump' and 'show' command 
Yes, unfortunately so, but that is something of the original architectural limitation.